pootle/apps/pootle_app/management/commands/update_tmserver.py

```diff
@@ -0,0 +1,159 @@
+                    default=False,
+                    help='Drop the entire index on start and update '
+                         'everything from scratch'),
+        make_option('--dry-run',
+                    action="store_true",
+                    dest='dry_run',
+                    default=False,
+                    help='Report only the number of translations to index '
+                         'and quit'),
+    )
+
+    def handle(self, *args, **options):
+        if not getattr(settings, 'POOTLE_TM_SERVER', False):
+            raise CommandError("POOTLE_TM_SERVER is missing from your settings.")
+        INDEX_NAME = settings.POOTLE_TM_SERVER['default']['INDEX_NAME']
+        es = Elasticsearch([{
+            'host': settings.POOTLE_TM_SERVER['default']['HOST'],
+            'port': settings.POOTLE_TM_SERVER['default']['PORT']
+        },
+        ])
+
+        last_indexed_revision = -1
+
+        if options['rebuild'] and not options['dry_run']:
+            if es.indices.exists(INDEX_NAME):
+                es.indices.delete(index=INDEX_NAME)
+
+        if not options['rebuild'] and not options['overwrite'] and es.indices.exists(INDEX_NAME):
+            result = es.search(
+                index=INDEX_NAME,
+                body={
+                    'query': {
+                        'match_all': {}
+                    },
+                    'facets': {
+                        'stat1': {
+                            'statistical': {
+                                'field': 'revision'
+                            }
+                        }
+                    }
+                }
+            )
+            last_indexed_revision = result['facets']['stat1']['max']
+
+        self.stdout.write("Last indexed revision = %s" % last_indexed_revision)
+
+
+        units_qs = Unit.simple_objects \
+            .exclude(target_f__isnull=True) \
+            .exclude(target_f__exact='') \
+            .filter(revision__gt=last_indexed_revision) \
+            .select_related(
+                'submitted_by',
+                'store',
+                'store__translation_project__project',
+                'store__translation_project__language'
+            ).values(
+                'id',
+                'revision',
+                'source_f',
+                'target_f',
+                'submitted_by__username',
+                'submitted_by__full_name',
+                'submitted_by__email',
+                'store__translation_project__project__fullname',
+                'store__pootle_path',
+                'store__translation_project__language__code'
+            ).order_by()
+
+        total = len(units_qs)
+
+        if not total:
+            self.stdout.write("No translations to index")
+            sys.exit()
+
+        self.stdout.write("%s translations to index" % total)
+
+        if options['dry_run']:
+            sys.exit()
+
+        i = 0
+        for unit in units_qs:
```

Comment:
`for index, unit in enumerate(units_qs):` is the Pythonic way do express this ;)
Also, use `units_qs.iterator()` to avoid loading the entire result set in memory.